python/lsst/jointcal/utils.py

```diff
@@ -0,0 +1,368 @@
+            # TODO: add CCD bounding boxes to plot once DM-5503 is finished.
+            # TODO: add a circle for the full focal plane.
+        length = (0.1*u.arcsecond).to(u.radian).value
+        ax.quiverkey(Q, 0.9, 0.95, length, '0.1 arcsec', coordinates='figure', labelpos='W')
+        plt.xlabel('RA')
+        plt.ylabel('Dec')
+        plt.title('visit: {}'.format(cat))
+        plt.savefig('.plots/{}-{}-quivers.pdf'.format(name, cat))
+
+
+def plot_wcs_quivers(ax, wcs1, wcs2, dim):
+    """Plot the delta between wcs1 and wcs2 as vector arrows."""
+
+    x1, y1, x2, y2 = make_xy_wcs_grid(dim, wcs1, wcs2)
+    uu = x2 - x1
+    vv = y2 - y1
+    return ax.quiver(x1, y1, uu, vv, units='x', pivot='tail', scale=1e-3, width=1e-5)
+
+
+def plot_wcs_magnitude(plt, data_refs, visit_catalogs, old_wcs_list, name):
+    for cat in visit_catalogs:
+        fig = plt.figure()
+        fig.set_tight_layout(True)
+        ax = fig.add_subplot(111)
+        xmin = np.inf
+        ymin = np.inf
+        xmax = -np.inf
+        ymax = -np.inf
+        for old_wcs, ref in zip(old_wcs_list, data_refs):
+            if ref.dataId['visit'] != cat:
+                continue
+            md = ref.get('calexp_md')
+            x1, y1, x2, y2 = make_xy_wcs_grid((md.get('NAXIS1'), md.get('NAXIS2')),
+                                              old_wcs, ref.get('wcs').getWcs())
+            uu = x2 - x1
+            vv = y2 - y1
+            extent = (x1[0, 0], x1[-1, -1], y1[0, 0], y1[-1, -1])
+            xmin = x1.min() if x1.min() < xmin else xmin
+            ymin = y1.min() if y1.min() < ymin else ymin
+            xmax = x1.max() if x1.max() > xmax else xmax
+            ymax = y1.max() if y1.max() > ymax else ymax
+            magnitude = (np.linalg.norm((uu, vv), axis=0)*u.radian).to(u.arcsecond).value
+            img = ax.imshow(magnitude, vmin=0, vmax=0.3,
+                            aspect='auto', extent=extent, cmap=plt.get_cmap('magma'))
+            # TODO: add CCD bounding boxes to the plot once DM-5503 is finished.
+            # TODO: add a circle for the full focal plane.
+
+        # We're reusing only one of the returned images here for colorbar scaling,
+        # but it doesn't matter because we set vmin/vmax so they are all scaled the same.
+        cbar = plt.colorbar(img)
+        cbar.ax.set_ylabel('distortion (arcseconds)')
+        plt.xlim(xmin, xmax)
+        plt.ylim(ymin, ymax)
+        plt.xlabel('RA')
+        plt.ylabel('Dec')
+        plt.title('visit: {}'.format(cat))
+        plt.savefig('.plots/{}-{}-heatmap.pdf'.format(name, cat))
+
+
+def plot_wcs(plt, wcs1, wcs2, dim, center=(0, 0), name=""):
+    """Plot the "distortion map": wcs1-wcs2 delta of points in the CCD grid."""
+
+    plt.figure()
+
+    x1, y1, x2, y2 = make_xy_wcs_grid(dim, wcs1, wcs2, num=50)
+    plt.plot((x1 - x2) + center[0], (y1 - y2) + center[1], '-')
+    plt.xlabel('delta RA (arcsec)')
+    plt.ylabel('delta Dec (arcsec)')
+    plt.title(name)
```

Comment:
This was a earlier plot with more limited utility, but I've added the option to save it. Since the code is there already, it can't hurt.